File: Golang/203_RemoveLinkedListElements.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeElements(head *ListNode, val int) *ListNode {
    dummy := new(ListNode)
    dummy.Next = head    
    p := dummy
    
    for p.Next != nil {
        if (p.Next.Val == val) {
            p.Next = p.Next.Next
        } else {
            p = p.Next
        }
    }
    
    return dummy.Next
}

File: C++/203_RemoveLinkedListElements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode dummy(0);
        dummy.next = head;
        ListNode* notEqual = &dummy;
        
        while (notEqual->next != NULL) {
            if (notEqual->next->val == val) {
                notEqual->next = notEqual->next->next;
            } else {
                notEqual = notEqual->next;
            }
        }
        return dummy.next;
    }
};
